File: Data Structures & Algorithms/is-anagram/submission-4.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int sn = s.length();
        int tn = t.length();
        int[] hash = new int[26];

        if(sn != tn) return false;
        for(int i = 0; i < sn; i++){
            hash[s.charAt(i) - 'a']++;
            hash[t.charAt(i) - 'a']--;
        }

        for(int i = 0; i < hash.length; i++){
            if(hash[i] != 0) return false;
        }
        return true;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        n = len(s)
        hash = set()
        left = 0
        longest = 0
        
        for right in range(n):
            c = s[right]
            while c in hash:
                hash.remove(s[left])
                left += 1
            hash.add(c)
            longest = max(longest, right-left+1)
        
        return longest